qemu/tests/qemu_guest_agent.py

```diff
@@ -809,6 +809,34 @@ def _action_after_fsfreeze(self, *args):
         else:
             self.test.fail("FS is not frozen,still can write in guest.")
 
+    def gagent_check_reset_reboot_shutdown(self, test, params, env):
+
+        """
+        Execute "system_reset" and send "shutdown,reboot"
+        command to guest agent after FS freezed
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        gagent = self.gagent
+        gagent.fsfreeze()
+
+        for mode in (gagent.SHUTDOWN_MODE_POWERDOWN, gagent.SHUTDOWN_MODE_REBOOT):
+            try:
+                if gagent.shutdown(mode):
+                    test.fail("agent shutdown command shouldn't succeed for freeze FS")
+            except guest_agent.VAgentCmdError as detail:
+                if not re.search('guest-shutdown has been disabled', str(detail)):
+                    test.fail(str(detail))
```

Comment:
Per https://github.com/avocado-framework/avocado-vt/pull/1391 You dropped `return True`, so should not judge it with `if gagent.shutdow()`. Here should be:
```
try:
    gagent.shutdown(mode)
except guest_agent.VAgentCmdError as detail:
    if not re.search('guest-shutdown has been disabled', str(detail)):
        test.fail(str(detail))
else:
    test.fail("agent shutdown command shouldn't succeed for freeze FS")